components/layout_2020/table/layout.rs

```diff
@@ -1327,27 +1348,33 @@ impl<'a> TableLayout<'a> {
     fn compute_table_height_and_final_row_heights(
         &mut self,
         mut row_sizes: Vec<Au>,
+        containing_block_for_children: &ContainingBlock,
         containing_block: &ContainingBlock,
     ) {
         // The table content height is the maximum of the computed table height from style and the
         // sum of computed row heights from row layout plus size from borders and spacing.
-        let table_height_from_style = self
-            .table
-            .style
-            .content_box_size(containing_block, &self.pbm)
-            .block
-            .auto_is(Length::zero)
-            .into();
+        let style = &self.table.style;
+        let table_height_from_style =
+            match style.content_box_size(containing_block, &self.pbm).block {
+                LengthPercentage(_) => containing_block_for_children.block_size,
+                Auto => style
+                    .content_min_box_size(containing_block, &self.pbm)
+                    .block
+                    .map(Au::from),
+            }
+            .auto_is(Au::zero);
```

Comment:
```suggestion
        // When there is a specified length for block size, `containing_block_for children` will have
        // this value, properly clamping for minimum and maximum block size.
        let table_height_from_style =
            match style.content_box_size(containing_block, &self.pbm).block {
                LengthPercentage(_) => containing_block_for_children.block_size,
                Auto => style
                    .content_min_box_size(containing_block, &self.pbm)
                    .block
                    .map(Au::from),
            }
            .auto_is(Au::zero);
```

Reply:
Tweaked the working a bit because it's not just when specifying lengths, also with percentages.